alica_engine/src/engine/RunningPlan.cpp

```diff
@@ -407,6 +408,11 @@ void RunningPlan::adaptAssignment(const RunningPlan& replacement)
         clearChildren();
         addChildren(replacement.getChildren());
         reactivate = true;
+        if (_ae->getTraceFactory() && _basicPlan && _basicPlan->getTraceContext().has_value()) {
```

Comment:
can remove the check for trace context. This method is only meant to be called by the scheduler thread. Can also remove the check for trace factory & do it in the basic plan's notify task assignment method instead so that it is more consistent with the rest of the tracing code.